src/main/remote-sync/RemoteSyncManager.test.ts

```diff
@@ -0,0 +1,284 @@
+import { DatabaseAdapter } from 'main/database/adapters/base';
+import { RemoteSyncManager } from './RemoteSyncManager';
+import { RemoteSyncedFile, RemoteSyncedFolder } from './helpers';
+import { sleep } from '../util';
+
+const inMemorySyncedFilesAdapter: DatabaseAdapter<RemoteSyncedFile> = {
+  get: jest.fn(),
+  connect: jest.fn(),
+  update: jest.fn(),
+  create: jest.fn(),
+  remove: jest.fn(),
+};
+
+const inMemorySyncedFoldersAdapter: DatabaseAdapter<RemoteSyncedFolder> = {
+  get: jest.fn(),
+  connect: jest.fn(),
+  update: jest.fn(),
+  create: jest.fn(),
+  remove: jest.fn(),
+};
+
+const createRemoteSyncedFileFixture = (
+  payload: Partial<RemoteSyncedFile>
+): RemoteSyncedFile => {
+  const result: RemoteSyncedFile = {
+    plainName: `file_${Date.now()}`,
+    id: Date.now(),
+    fileId: Date.now().toString(),
+    type: 'jpg',
+    size: '999',
+    bucket: `bucket_${Date.now()}`,
+    folderId: 555,
+    folderUuid: '',
+    encryptVersion: '03-aes',
+    userId: 567,
+    modificationTime: new Date().toISOString(),
+    createdAt: new Date().toISOString(),
+    updatedAt: new Date().toISOString(),
+    ...payload,
+  };
+
+  return result;
+};
+
+const createRemoteSyncedFolderFixture = (
+  payload: Partial<RemoteSyncedFolder>
+): RemoteSyncedFolder => {
+  const result: RemoteSyncedFolder = {
+    plainName: `folder_${Date.now()}`,
+    id: Date.now(),
+    type: 'folder',
+    bucket: `bucket_${Date.now()}`,
+    userId: 567,
+    createdAt: new Date().toISOString(),
+    updatedAt: new Date().toISOString(),
+    parentId: null,
+    uuid: `xxxxx-xxxxx-${Date.now()}`,
```

Comment:
Could we use `v4()` from 'uuid' to generate UUIDs and therefore, make it nearer to the real cases?